spring-data-couchbase/src/main/java/org/springframework/data/couchbase/core/convert/AbstractCouchbaseConverter.java

```diff
@@ -93,39 +98,70 @@ public void afterPropertiesSet() {
 		conversions.registerConvertersIn(conversionService);
 	}
 
+	/**
+	 * This convertForWriteIfNeeded takes a property and accessor so that the annotations can be accessed (ie. @Encrypted)
+	 *
+	 * @param prop the property to be converted to the class that would actually be stored.
+	 * @param accessor the property accessor
+	 * @return
+	 */
 	@Override
-	public Object convertForWriteIfNeeded(Object value) {
+	public Object convertForWriteIfNeeded(CouchbasePersistentProperty prop, ConvertingPropertyAccessor<Object> accessor,
+			boolean processValueConverter) {
+		Object value = accessor.getProperty(prop, prop.getType());
 		if (value == null) {
 			return null;
 		}
+		if (processValueConverter && conversions.hasValueConverter(prop)) {
+			CouchbaseDocument encrypted = (CouchbaseDocument) conversions.getPropertyValueConversions()
+					.getValueConverter(prop)
+					.write(value, new CouchbaseConversionContext(prop, (MappingCouchbaseConverter) this, accessor));
+			return encrypted;
+		}
+		Class<?> targetClass = this.conversions.getCustomWriteTarget(value.getClass()).orElse(null);
 
-		return this.conversions.getCustomWriteTarget(value.getClass()) //
-				.map(it -> (Object) this.conversionService.convert(value, it)) //
+		boolean canConvert = targetClass == null ? false
+				: this.conversionService.canConvert(new TypeDescriptor(prop.getField()), TypeDescriptor.valueOf(targetClass));
+		if (canConvert) {
+			return this.conversionService.convert(value, new TypeDescriptor(prop.getField()),
+					TypeDescriptor.valueOf(targetClass));
+		}
+
+		Object result = this.conversions.getCustomWriteTarget(prop.getType()) //
+				.map(it -> this.conversionService.convert(value, new TypeDescriptor(prop.getField()),
+						TypeDescriptor.valueOf(it))) //
 				.orElseGet(() -> Enum.class.isAssignableFrom(value.getClass()) ? ((Enum<?>) value).name() : value);
 
+		return result;
+
 	}
 
-	/* TODO needed later
```

Comment:
The 'diff' gets confusing here.  I removed a commented-out convertToCouchbaseType() method here and added a new convertForWriteIfNeeded() (above) that has three args vs. the existing one that had only one arg (below).